### PR TITLE
Aokana extraction support

### PR DESCRIPTION
**Pull request by Cqfuj**
Aokana datas are stored with another layer of encryption: bse files. They are composed of a small header followed by the file with only its first 64 bytes encrypted.
The decryption algorithm is taken from the [arc-reader](https://github.com/minirop/arc-reader) repository. I just translated it to c++ and added small organization changes.
Tell me what you think.

### CODE REVIEW

File: src/dec/bgi/bse_file_decoder.cc [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        auto i = rand() & 0x3F;
+        while (flags[i])
+            i = (i + 1) & 0x3F;
+
+        const auto shift = rand() & 0x07;
+        const auto key   = rand();
+        const auto data  = ((this->data[i] & 0xFF) - rand()) & 0xFF;
+
+        if (key & 1)
+            this->data[i] = data << shift | data >> (8 - shift);
```
Comment (rr-) [inline src/dec/bgi/bse_file_decoder.cc]: Consider using `algo::rotl` and `algo::rotr`

File: src/dec/bgi/bse_file_decoder.cc [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    input_file.stream.skip(2);
+    check_sum = input_file.stream.read<u8>();
+    check_xor = input_file.stream.read<u8>();
+    seed = input_file.stream.read_le<s32>();
+    data = input_file.stream.read(data_size);
+}
+
+s32 BseDecoder::rand()
+{
+    const auto tmp = (((seed * 257 >> 8) + seed * 97) + 23) ^ -1496474763;
```
Comment (rr-) [inline src/dec/bgi/bse_file_decoder.cc]: Please encode -1496474763 as a positive number in hex, like `0xDEADBEEF` (I believe it's `0xA6CD9B75` in this particular case)

File: src/dec/bgi/bse_file_decoder.cc [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+namespace
+{
+    class BseDecoder final
+    {
+    public:
+        BseDecoder(io::File &input_file);
+        bstr decode();
+
+        const u32 data_size = 64;
+    private:
```
Comment (rr-) [inline src/dec/bgi/bse_file_decoder.cc]: Nitpick - please put a newline above `private`

**Comment by rr-** [comment]
I like it, very nice :+1: BTW for the file formats in `GAMELIST.js` you can use `./tools/analyze` and check for "recognized as" entries.

**Comment by Cqfuj** [comment]
Thanks, I followed your recomendations:
* add a new line above `private`
* use of `algo::rotl` and `algo::rotr` in `decode()`
* change -1496474763 to 0xA6CD9B75

I also did other few changes:
* removed some bitwise _and_ where they were not needed
* replaced bitwise _and_ by modulo operator when computing the shift and the index
* changed 32767 to 0x7FFF in `rand()` 
* used a rotation in `rand()`

**Comment by rr-** [comment]
Merged, thanks.

**Pull request merged**